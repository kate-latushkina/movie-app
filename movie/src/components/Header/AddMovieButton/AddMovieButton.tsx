import React from "react";
import { Button } from "@material-ui/core";

const AddMovieButton: React.FC = () => {
  return (
    <Button variant="outlined" color="secondary">
      + Add movie
    </Button>
  );
};

export default AddMovieButton;
