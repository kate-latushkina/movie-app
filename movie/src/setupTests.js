// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow, render, mount } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

global.shallow = shallow;
global.render = render;
global.mount = mount;
global.toJson = toJson;

// Fail tests on any warning
console.error = message => {
  throw new Error(message);
};
