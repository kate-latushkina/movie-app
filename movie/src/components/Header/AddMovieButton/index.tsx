import AddMovieButton from "./AddMovieButton";

export default AddMovieButton;
