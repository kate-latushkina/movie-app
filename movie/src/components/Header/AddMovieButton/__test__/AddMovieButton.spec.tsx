import React from "react";
import { Button } from "@material-ui/core";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import AddMovieButton from "../AddMovieButton";
import "../../../../setupTests";

describe("should render AddMovieButton component", () => {
  it("should contain Button wrapper", () => {
    const component = shallow(<AddMovieButton />);
    const wrapper = component.find(Button);
    expect(wrapper.length).toBe(1);
  });

  it("should render AddMovieButton snapshot", () => {
    const wrapper = renderer.create(<AddMovieButton />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
