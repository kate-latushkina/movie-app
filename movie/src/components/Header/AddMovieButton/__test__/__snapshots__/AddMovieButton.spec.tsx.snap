// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddMovieButton component should render AddMovieButton snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    + Add movie
  </span>
</button>
`;
